src/App.tsx

```diff
@@ -1,26 +1,157 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import { Header } from './copmonents/Header/Header';
+import { TodoList } from './copmonents/TodoList/TodoList';
+import { Footer } from './copmonents/Footer/Footer';
+import { Todo } from './types/Todo';
+import * as todoMethods from './api/todos';
+import { ErrorNotification } from './copmonents/Error/Error';
+import { FilterStatus } from './types/FilterStatus';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterStatus, setFilterStatus] = useState<FilterStatus>('all');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletingTodoIds, setDeletingTodoIds] = useState<number[] | null>(null);
+  const [isInputDisabled, setInputDisabled] = useState(false);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isInputDisabled]);
+
+  useEffect(() => {
+    todoMethods
+      .getTodos()
+      .then(setTodos)
+      .catch(error => {
+        setErrorMessage('Unable to load todos');
+        throw error;
+      });
+  }, []);
+
+  const filteredTodos = useMemo((): Todo[] => {
+    if (filterStatus === 'all') {
+      return todos;
+    }
+
+    if (filterStatus === 'active') {
+      return todos.filter(todo => !todo.completed);
+    }
+
+    return todos.filter(todo => todo.completed);
```

Comment:
Consider will be better if you use switch case here, also, create a enum for 'all', 'active', 'completed' and use it everywhere